packages/nimble-components/src/table/specs/table-column-placeholder-hld.md

```diff
@@ -0,0 +1,200 @@
+| Non-mapped value\*         | \<empty cell> | \<empty group row> |
+
+\*This is considered invalid data from the table's perspective and should be fixed within the client application.
+
+Column best practices:
+
+-   Avoid mixing `undefined` and `null` as values for the same field. When grouping this will lead to two groups (one for `null` values and one for `undefined` values) that both have the text `"No value"`.
+-   Avoid using values that do not correspond to a mapping for the column
+
+### Implementation plan
+
+A column's placeholder will be stored as part of that column's `columnConfig` object. There will be no changes to the `TableColumn`, `ColumnInternals`, or `ColumnInternalsOptions` classes. The placeholder from the `columnConfig` object will be used by the cell views when rendering the cell.
+
+We will create a placeholder mixin that adds the following to columns that chose to use it:
+
+-   `placeholder` string property
+-   `placeholder` attribute
+-   abstract `placeholderChanged` function that will force columns using the mixin to implement `placeholderChanged` to update their column configuration
+
+The columns that will be updated to use this mixin are:
+
+-   TableColumnText
+-   TableColumnAnchor
+-   TableColumnNumberText
+-   TableColumnDateText
+-   TableColumnDurationText
+
+### Localization
+
+All group row placeholder strings will be localized through the table's localization provider. Those strings are:
+
+-   No value
+-   Empty
+
+If an application is localized, it can set a column's `placeholder` to a localized value.
+
+## Alternative Implementations / Designs
+
+### Configurable placeholders in group rows
+
+In addition to extending column APIs to have a `placeholder` property, they could also be extended to have a `group-placeholder` property. However, this level of configuration is not required by any client applications today. This feature is also likely to introduce inconsistency throughout our applications.
+
+If an application wants to modify the value of a group row placeholder without this feature, they can do so through the localization provider.
+
+### Allow different placeholders to be configured for each cell
+
+We could add the ability to have different placeholders configured for each cell, but this poses a few different problems:
+
+1. This would likely need to be a drastically different API, such as having the placeholder specified in the record. That would lead to quite a bit of duplicate information being set on the table, particularly when there are no use cases for this right now.
+1. This would lead to a confusing state for the user because the placholder would be different for various rows, but all those rows would be in a single group.
+
+A use case for different information being presented to the user for each cell will likely be solved by [a different feature to show cell-specific state](https://github.com/ni/nimble/issues/1776).
+
+## Future Work
```

Comment:
Anything we could do to help us think about placeholder support when we add new column types? For example,
to help us decide what to do for new columns we could add a section that records the philosophies you used to generate the column-specific guidance. Something like:
    - columns that display text should typically have a placeholder, columns that don't shouldn't
    - *whatever exception to the above that convinced us enum-text didn't need a placehoder*

If it's really use-case specific and not generalizable then feel free to reject this idea.